src/parameterizations/lateral/MOM_lateral_mixing_coeffs.F90

```diff
@@ -1128,6 +1153,123 @@ subroutine calc_slope_functions_using_just_e(h, G, GV, US, CS, e)
+  !real :: dz_tot(SZI_(G),SZJ_(G)) ! The total thickness of the water columns [Z ~> m]
+  ! real :: dz(SZI_(G),SZJ_(G),SZK_(GV)) ! The vertical distance across each layer [Z ~> m]
+  real :: Ux_Hx(SZIB_(G), SZJ_(G))  ! X-slope of U and H  [T-1 ~> s-1]
+  real :: Uy_Hy(SZI_(G), SZJB_(G))  ! Y-slope of U and H  [T-1 ~> s-1]
+  real :: Vx_Hx(SZIB_(G), SZJ_(G))  ! X-slope of V and H  [T-1 ~> s-1]
+  real :: Vy_Hy(SZI_(G), SZJB_(G))  ! Y-slope of V and H  [T-1 ~> s-1]
+  real :: H_cutoff      ! Local estimate of a minimum thickness for masking [H ~> m or kg m-2]
+  !real :: dZ_cutoff     ! A minimum water column depth for masking [H ~> m or kg m-2]
+  real :: h_neglect     ! A thickness that is so small it is usually lost
+                        ! in roundoff and can be neglected [H ~> m or kg m-2].
+  real :: graduh        ! Gradient model frequency, zonal transport [T-1 ~> s-1]
+  real :: gradvh        ! Gradient model frequency, merid transport [T-1 ~> s-1]
+  real :: Hup, Hdn      ! Thickness from above, below [H ~> m or kg m-2]
+  real :: H_geom        ! The geometric mean of Hup*Hdn [H ~> m or kg m-2].
+  !logical :: use_dztot  ! If true, use the total water column thickness rather than the
+                        ! bathymetric depth for certain calculations.
+  real    :: UH_grad_local(SZIB_(G), SZJ_(G),SZK_(GV))  ! The depth integral of grad slopes for UH at u-points
+  real    :: VH_grad_local(SZI_(G), SZJB_(G),SZK_(GV))  ! The depth integral of grad slopes for VH at v-points
+  real    :: Lgrid      ! Grid lengthscale for the gradient model [H ~> m]
+  integer :: is, ie, js, je, nz
+  integer :: i, j, k
+  integer :: l_seg
+
+  if (.not. CS%initialized) call MOM_error(FATAL, "calc_slope_functions_with_gradient_model: "// &
+         "Module must be initialized before it is used.")
+
+  if (.not. CS%use_gradient_model) return
+  if (.not. allocated(CS%UH_grad)) call MOM_error(FATAL, "calc_slope_function:"// &
+         "UH_grad is not associated with use_gradient_model.")
+  if (.not. allocated(CS%VH_grad)) call MOM_error(FATAL, "calc_slope_function:"// &
+         "VH_grad is not associated with use_gradient_model.")
+
+  is = G%isc ; ie = G%iec ; js = G%jsc ; je = G%jec ; nz = GV%ke
+
+  h_neglect = GV%H_subroundoff
+  H_cutoff = real(2*nz) * (GV%Angstrom_H + h_neglect)
+
+
+  ! Set the length scale at u-points for the gradient model
+  do j=js,je ; do I=is-1,ie
+    Lgrid = sqrt(G%dxCu(I,j)**2 + G%dyCu(I,j)**2)
+    CS%L2grad_u(I,j) = 1.0 * Lgrid**2
+  enddo ; enddo
+  ! Set length scale at v-points for the gradient model
+  do J=js-1,je ; do i=is,ie
+    Lgrid = sqrt(G%dxCv(i,J)**2 + G%dyCv(i,J)**2)
+    CS%L2grad_v(i,J) = 1.0 * Lgrid**2
+  enddo ; enddo
+
+  do k=nz,CS%VarMix_Ktop,-1
+
+    ! Calculate the gradient slopes Ux_Hx, Vx_Hx, Uy_Hy, Vy_Hy on u- and v-points respectively
+    do j=js-1,je+1 ; do I=is-1,ie
+      Ux_Hx(I,j) = ((G%IdxCu(I+1,j)*G%IdyCu(I+1,j)*uh(I+1,j,K)) - (G%IdxCu(I,j)*G%IdyCu(I,j)*uh(I,j,k))) * ( &
```

Comment:
This expression will not pass rotational symmetry testing.  The first part of the numerator in this expression is staggered at the same point as `h(i+1,j)`, while `(G%IareaT(I+1,j) + G%IareaT(I,j))` and the normalized thickness gradient is staggered at the same point as `u(I,j)`, and `G%dyT(i,j)` is staggered at the same point as `h(i,j)`.

You can verify the rotational consistency of these new code by doing a small single-PE run that exercises it with the run-time settings `ROTATE_INDEX = True` and `INDEX_TURNS` set to 0, 1, 2, or 3.  If the expressions are rotationally consistent, you will get identical answers in each case.  If the answers change (as I suspect they will here) you will get different answers.